### ISSUE TITLE
Improve workflow to share and consume common project templates

### ISSUE THREAD

**Issue by danfhernandez** [pulumi/pulumi#4835]
A simple solution to this could be to "load" templates from a repo into `.pulumi/templates` with an additional CLI command. 

A command like `pulumi templates add [git repo url]` would add the additional templates to the local installation and updates a config file in `.pulumi` that points to that repo. That config file would contain a list of pointers to repos for templates to pull in and would default to a single entry pointing to https://github.com/pulumi/templates.  

You could then get updates with a command like `pulumi templates update`.  

This would actually also make it possible to replace the default list of templates that may be more relevant to an organization.

**Comment by danfhernandez** [comment]
Random additional side benefit is that it solves the problem loading in templates in environments that don't allow cloning of public repos (which I know is an unlikely case but actually happened at Morgan Stanley). You could create a repo accessible within the network and point to that for your templates.

**Comment by lukehoban** [comment]
A simple solution here might be to just have an env var that overrides the default repo to pull templates from.

**Comment by danfhernandez** [comment]
Yeah that's definitely the simplest solution to do a "replace" and actually probably a good place to start that covers an org that wants to replace the default list and the "no access to public repo" problem.  

I think "adding" is more valuable from a day-to-day usage perspective. That allows anyone to add additional templates from various resources and even subscribe to others that could be useful. Feels almost like a simplified version of the "gallery" idea.

**Comment by danfhernandez** [comment]
And then when you have these "template packs" or "template repos".. whatever that name would be.. when you run `pulumi up` you could include the named version of that like `pulumi new k8s-examples` and you would get the list of k8s examples from the k8s template repo that you've added to pick from.  

I feel like I can keep going but I'll stop... :)

**Comment by danfhernandez** [comment]
I just had a realization that what I was describing above is essentially how homebrew taps work... i.e. the following workflow: 
1. `pulumi tap pulumi/examples`
> Obviously rename tap to something else... maybe `pulumi template-repo add pulumi/examples` 
2. `pulumi new pulumi/examples/aws-cs-webserver`

Other possible commands: 
- `pulumi template ls` 
- `pulumi template-repo ls` 
- `pulumi template rm` 
- `pulumi template-repo-rm` 
- etc

**Comment by jkodroff** [comment]
I really like this idea. This would also allow the community, customers, and partners to all easily enable their own templates from the command line, all while reducing (or at least not increasing) our maintenance burden for `pulumi/templates`.
